policy/scan/local_scanner.go

```diff
@@ -462,6 +479,43 @@ func (s *LocalScanner) HealthCheck(ctx context.Context, req *HealthCheckRequest)
 	}, nil
 }
 
+func (s *LocalScanner) getUpstreamConfig(incognito bool, job *Job) (resources.UpstreamConfig, error) {
+	if incognito {
+		return resources.UpstreamConfig{Incognito: true}, nil
+	}
+
+	pluginsMap := s.pluginsMap
```

Comment:
Have you checked that this has no side-effect? I think we need to copy the keys, otherwise you manipulate the `s.pluginsMap` with the one-off credentials

Reply:
Good catch, this is now a copied map so we do not touch the original one